### PR TITLE
Mitigate actor placement race condition on leadership change

### PR DESCRIPTION
**Pull request by artursouza**
# Description

This PR is a redo of https://github.com/dapr/dapr/pull/7312 since it was accidentally merged by a wrong push to upstream by me.

## Issue reference

Mitigates (not the permanent fix yet): #7311
## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by yaron2 — DISMISSED** [review]

File: pkg/placement/membership.go [hunk 1]
```diff
@@ -176,6 +176,9 @@ func (p *Service) cleanupHeartbeats() {
 // membershipChangeWorker is the worker to change the state of membership
 // and update the consistent hashing tables for actors.
 func (p *Service) membershipChangeWorker(ctx context.Context) {
+	baselineHeartbeatTimestamp := p.clock.Now().UnixNano()
+	log.Infof("Baseline membership heartbeat timestamp: %v", baselineHeartbeatTimestamp)
```
[the remark this answers is not in the document]
Comment (artursouza) [reply pkg/placement/membership.go:180]: I thought about it but this log only happens on leadership change and is very helpful to troubleshoot. Given this is not as verbose and we are still troubleshooting issues with placement, I made this Info for now.

**Review by mukundansundar — APPROVED** [review]

**Comment by artursouza** [comment]
Integration tests are failing at 100% and it seems totally unrelated to my change.

**Comment by artursouza** [comment]
I suggest we merge this PR or have a different PR first to make integration tests stable in release-1.12 branch.

**Review by daixiang0 — APPROVED** [review]
LGTM

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**